src/creature.cpp

```diff
@@ -139,27 +139,27 @@ void Creature::onThink(uint32_t interval)
 		blockTicks = 0;
 	}
 
-	if (followCreature) {
-		walkUpdateTicks += interval;
-		if (forceUpdateFollowPath || walkUpdateTicks >= 2000) {
-			walkUpdateTicks = 0;
-			forceUpdateFollowPath = false;
-			isUpdatingPath = true;
-		}
-	}
-
-	if (isUpdatingPath) {
-		isUpdatingPath = false;
-		goToFollowCreature();
-	}
-
 	// scripting event - onThink
 	const CreatureEventList& thinkEvents = getCreatureEvents(CREATURE_EVENT_THINK);
 	for (CreatureEvent* thinkEvent : thinkEvents) {
 		thinkEvent->executeOnThink(this, interval);
 	}
 }
 
+void Creature::forceUpdatePath()
+{
+	if (!attackedCreature && !followCreature) {
+		return;
+	}
+
+	if (lastPathUpdate - OTSYS_TIME() > 0) {
+		return;
+	}
+
+	lastPathUpdate = OTSYS_TIME() + 200;
```

Comment:
Here I would replace the "200" with EVENT_CREATURE_PATH_INTERVAL instead, or even possibly give it its own variable that would also be configurable, but I think one variable for both things should suffice.

Reply:
We don't really want to have them linked so having a secondary variable would be better. The idea here is just to stop paths from updating too fast, which having a 200ms delay is a pretty good amount of time that doesn't make the monsters seem like they are slow.